src/test/java/htsjdk/samtools/cram/compression/rans/RansTest.java

```diff
@@ -49,107 +50,173 @@ public String toString() {
         };
     }
 
-    @Test(dataProvider="ransData")
-    public void testRANS(final TestCaseWrapper tc) {
-        roundTripForEachOrder(tc.testArray);
+    @DataProvider(name="ransCodecs")
+    public Object[][] getRansCodecs() {
+        final RANS4x8 rans4x8 = new RANS4x8();
+        final RANSNx16 ransNx16 = new RANSNx16();
+        final RANS4x8Params rans4x8ParamsOrder0 = new RANS4x8Params(RANSParams.ORDER.ZERO); // RANS4x8 Order 0
+        final RANS4x8Params rans4x8ParamsOrder1 = new RANS4x8Params(RANSParams.ORDER.ONE); // RANS4x8 Order 1
+        final RANSNx16Params ransNx16ParamsFormat0 = new RANSNx16Params(0); //RANSNx16 Format(first byte) 0
+        // TODO: More Format values i.e, combinations of bit flags will be added later
+        return new Object[][]{
+                {rans4x8, rans4x8ParamsOrder0},
+                {rans4x8, rans4x8ParamsOrder1},
+                {ransNx16, ransNx16ParamsFormat0}
+        };
+    }
+
+    @DataProvider(name="ransCodecsAndData")
+    public Object[][] getRansCodecsAndData() {
+        // this data provider provides all the testdata for each of the RANS type and order
+        return TestNGUtils.cartesianProduct(getRansTestData(), getRansCodecs());
```

Comment:
It seems inconsistent that `getRansCodecs` is annotated as a data provider and `getRansTestData` isn't. They should both the the same (??), probably not annotated.

Reply:
Also, I do think that, given that the `compress` signature and the data types of the arguments to the `compress` methods are different for 4x8 and Nx16, we'll probably need to separate this dataprovider into two separate dataproviders, one that does the cartesian product of all of the 4x8s test cases (with varying orders, etc) with the test data, and one that does the cartesian product of Nx16s with the test data. Then you'll have to have different (separate) tests for those two data providers, each one of which will have arguments of the correct corresponding type.

Reply:
Actually, it more or less looks like thats what you have in the CRAMCorpus tests (separate providers and tests for 4x8 and Nx16).

Reply:
As of the next commit, I am annotating a method as a data provider only if it is used by at least one test as a data provider.